./build.sh --portablebulid false fails

I would like to add a build configuration in CI to help catch issues like https://github.com/dotnet/runtime/pull/42415. That is, essentially do a `./build.sh --portablebuild false` in a CI leg (on x64).

But that doesn't seem to work on Fedora or RHEL systems that I tried it on.

They all fail with a messages like

```
/home/dotnet/runtime/src/libraries/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.ilproj : error NU1101: Unable to find package runtime.rhel.8-x64.microsoft.netcore.ilasm. No packages exist with this id in source(s): dotnet-eng
```
It's complaining that the ilasm assets for non-portable RIDs can not be found.

Does a portable build work outside of source-build? Is there a way I can make the build use portable nuget packages for tools like ilasm?

--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -155,7 +155,9 @@
     <!-- There are no Mac Catalyst, iOS or tvOS tools and it can be built on OSX only, so use that -->
     <_toolRuntimeRID Condition="'$(_runtimeOS)' == 'maccatalyst' or '$(_runtimeOS)' == 'ios' or '$(_runtimeOS)' == 'iOSSimulator' or '$(_runtimeOS)' == 'tvos' or '$(_runtimeOS)' == 'tvOSSimulator'">osx-x64</_toolRuntimeRID>
 
-    <MicrosoftNetCoreIlasmPackageRuntimeId>$(_toolRuntimeRID)</MicrosoftNetCoreIlasmPackageRuntimeId>
+    <!-- There are no non-portable builds for Ilasm/Ildasm -->
+    <MicrosoftNetCoreIlasmPackageRuntimeId Condition="'$(PortableBuild)' != 'true' and '$(_portableOS)' == 'linux'">linux-$(_hostArch)</MicrosoftNetCoreIlasmPackageRuntimeId>
+    <MicrosoftNetCoreIlasmPackageRuntimeId Condition="'$(MicrosoftNetCoreIlasmPackageRuntimeId)' == ''">$(_toolRuntimeRID)</MicrosoftNetCoreIlasmPackageRuntimeId>
 
     <_packageRID Condition="'$(PortableBuild)' == 'true'">$(_portableOS)-$(TargetArchitecture)</_packageRID>
     <PackageRID Condition="'$(PackageRID)' == ''">$(_packageRID)</PackageRID>

--- a/eng/pipelines/global-build.yml
+++ b/eng/pipelines/global-build.yml
@@ -133,5 +133,5 @@ jobs:
     - SourceBuild_Linux_x64
     jobParameters:
       nameSuffix: SourceBuild
-      buildArgs: -subset clr+libs+host+packs /p:DotNetBuildFromSource=true
+      buildArgs: -subset clr+libs+host+packs /p:DotNetBuildFromSource=true --portableBuild false
       timeoutInMinutes: 90

--- a/src/libraries/Native/Unix/System.Security.Cryptography.Native/opensslshim.h
+++ b/src/libraries/Native/Unix/System.Security.Cryptography.Native/opensslshim.h
@@ -1069,6 +1069,7 @@ FOR_ALL_OPENSSL_FUNCTIONS
 #define EVP_CIPHER_CTX_free local_EVP_CIPHER_CTX_free
 #define EVP_CIPHER_CTX_new local_EVP_CIPHER_CTX_new
 #define EVP_CIPHER_CTX_reset local_EVP_CIPHER_CTX_reset
+#define EVP_PKEY_get0_RSA local_EVP_PKEY_get0_RSA
 #define EVP_PKEY_up_ref local_EVP_PKEY_up_ref
 #define HMAC_CTX_free local_HMAC_CTX_free
 #define HMAC_CTX_new local_HMAC_CTX_new

